accounting/transferwise/tests/test_mixins.py

```diff
@@ -0,0 +1,142 @@
+        self.payment1 = self._create_payment(recipient_bank_account_kwargs={
+            'transferwise_recipient_id': 1,
+            'user_account__business_name': 'Developer 1',
+            'identification': {
+                'account_number': 10,
+                'currency': 'PKR',
+            }
+        })
+        self.payment2 = self._create_payment(recipient_bank_account_kwargs={
+            'transferwise_recipient_id': 2,
+            'user_account__business_name': 'Developer 2',
+            'identification': {
+                'account_number': 11,
+                'currency': 'USD',
+            }
+        })
+        self.mixin = mixins.TransferWiseCsvMixin()
+        self.csv_path = ''
+        super().setUp()
+
+    def tearDown(self):
+        """ Delete any leftover files. """
+        if self.csv_path:
+            os.unlink(self.csv_path.name)
+        super().tearDown()
+
+    def test_to_bulk_payment_csv(self):
+        """ A bulk payment CSV is generated with all the proper columns and rows. """
+        self.csv_path = self.mixin.to_bulk_payment_csv([self.payment1, self.payment2])
+        with open(self.csv_path.name, newline='') as csv_file:
+            reader = csv.reader(csv_file)
+            self.assertEqual(next(reader), [
+                'recipientId',
+                'name',
+                'account',
+                'sourceCurrency',
+                'targetCurrency',
+                'amountCurrency',
+                'amount',
+                'paymentReference'
+            ])
+            self.assertEqual(next(reader), [
+                '1',
+                'Developer 1',
+                '10',
+                'EUR',
+                'PKR',
+                'EUR',
+                '100.0000000000',
+                '2018-01'
+            ])
+            self.assertEqual(next(reader), [
+                '2',
+                'Developer 2',
+                '11',
+                'EUR',
+                'USD',
+                'EUR',
+                '100.0000000000',
+                '2018-01'
+            ])
+            with self.assertRaises(StopIteration):
+                next(reader)
+
+    def test_to_bulk_payment_csv_no_payments(self):
+        """ A bulk payment CSV is generated with only columns and no rows because no payments exist. """
+        self.csv_path = self.mixin.to_bulk_payment_csv([])
+        with open(self.csv_path.name, newline='') as csv_file:
+            reader = csv.reader(csv_file)
+            self.assertEqual(next(reader), [
+                'recipientId',
+                'name',
+                'account',
+                'sourceCurrency',
+                'targetCurrency',
+                'amountCurrency',
+                'amount',
+                'paymentReference'
+            ])
+            with self.assertRaises(StopIteration):
+                next(reader)
+
+    def _create_payment(self, recipient_bank_account_kwargs=None):
```

Comment:
@UmanShahzad Nit: Move this above the test methods to make it easier to avoid mixing helper methods and test methods in the future.

Reply:
:+1: